### DISCUSSION TITLE
show applied RaplayGain somewhere

### DISCUSSION THREAD

**Discussion by Almoped** [audacious-media-player/audacious#102] [Ideas]
I could not see applied ReplayGain anywhere in the UI. I guess it could make sense to display in the track mouseover info, the right-click Song Info or the status bar. Using the following it can be shown in the status bar but duplicates replay_gain_mode code. Not sure I made the code tags work.

.../libaudcore/playback.cc
`
EXPORT void aud_drct_get_info(int & bitrate, int & samplerate, int & channels, float & rg)
{
    auto mh = mutex.take();
    bool ready = is_ready(mh);

    bitrate = ready ? pb_info.bitrate : 0;
    samplerate = ready ? pb_info.samplerate : 0;
    channels = ready ? pb_info.channels : 0;

    if (pb_info.gain_valid)
    {
        auto mode = (ReplayGainMode)aud_get_int("replay_gain_mode");
        if ((mode == ReplayGainMode::Album) ||
            (mode == ReplayGainMode::Automatic &&
             (!aud_get_bool("shuffle") || aud_get_bool("album_shuffle"))))
        {
            rg = pb_info.gain.album_gain;
        }
        else
        {
            rg = pb_info.gain.track_gain;
        }
    }
    else
        rg = 0;
}
`

.../qtui/status_bar.cc
`
void StatusBar::update_codec()
{
    /* codec info is hidden when a message is displayed */
    if (!currentMessage().isEmpty())
        return;

    if (!aud_drct_get_ready())
    {
        codec_label->hide();
        return;
    }

    Tuple tuple = aud_drct_get_tuple();
    String codec = tuple.get_str(Tuple::Codec);

    int bitrate, samplerate, channels;
    float rg;
    aud_drct_get_info(bitrate, samplerate, channels, rg);

    StringBuf buf(0);

    if (codec)
    {
        buf.insert(-1, codec);
        if (channels > 0 || samplerate > 0 || bitrate > 0)
            buf.insert(-1, ", ");
    }

    if (channels > 0)
    {
        if (channels == 1)
            buf.insert(-1, _("mono"));
        else if (channels == 2)
            buf.insert(-1, _("stereo"));
        else
            str_append_printf(
                buf, ngettext("%d channel", "%d channels", channels), channels);

        if (samplerate > 0 || bitrate > 0)
            buf.insert(-1, ", ");
    }

    if (samplerate > 0)
    {
        str_append_printf(buf, _("%d kHz"), samplerate / 1000);
        if (bitrate > 0)
            buf.insert(-1, ", ");
    }

    if (bitrate > 0)
    {
        str_append_printf(buf, _("%d kbit/s"), bitrate / 1000);
        if (rg != 0)
            buf.insert(-1, ", RG: ");
    }

    if (rg != 0)
        str_append_printf(buf, _("%.2f dB"), rg);

    codec_label->setText((const char *)buf);
    codec_label->show();
}
`

**Comment by Almoped** [comment]
Example
![qtui_statusbar](https://github.com/audacious-media-player/audacious/assets/21230170/31f5a504-0718-4964-a2e2-7709ee5d0edd)

**Comment by radioactiveman** [comment]
Thanks for the suggestion and patches.

I would prefer not to show replay gain info in the status bar. However it fits well into the song info dialog (libaudqt/info-dialog, info-widget). Could you please implement it there and submit a pull request? Thanks.
